### PR TITLE
allow ecbuild_git to checkout git submodules

### PR DESCRIPTION
**Pull request by aerorahul**
For some time now, [JCSDA](https://github.com/jcsda) has been forking third-party repositories. If there are git submodules employed by a repository that is used/co-developed, those submodules have to be forked as well.  Examples (relevant to JCSDA) are `mom6` and it's dependencies e.g. `CVmix`, `geoKdtree`, `mom6-da-hooks` and now `ocean_bgc`.
This adds burden on  the developer to maintain the `ecbuild` layer on these submodules as well as keeping them in sync; i.e. keep a track of the `SHA` in the submodules.

With this option `RECURSIVE`, one can recursively clone/update these via the `ecbuild_bundle`, this makes the job much easier and allows us to obtain the appropriate `SHA` of these dependencies as they relate to the upstream repository.  The appropriate git call `git submodule update --init --recursive` is called when this option is specified in the call to `ecbuild_bundle` or `ecbuild_git`. 

An example of this is demonstrated in `mom6-bundle`. Clone the branch `feature/recursion` from [mom6-bundle](https://github.com/aerorahul/mom6-bundle) and use the branch with the same name in `jcsda::ecbuild`.

Change-Id: I586486448540546e2f360efc25d78a1d1d4f34b6

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](http://claassistant.ecmwf.int:80/ecmwf/ecbuild?pullRequest=19) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](http://claassistant.ecmwf.int:80/ecmwf/ecbuild?pullRequest=19) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://claassistant.ecmwf.int:80/check/ecmwf/ecbuild?pullRequest=19) it.</sub>

**Comment by aerorahul** [comment]
Thank you @wdeconinck for adding the relevant reviewers.  Please let us know if there are any concerns/comments.

File: cmake/ecbuild_git.cmake [hunk 1]
```diff
@@ -224,6 +228,16 @@ macro( ecbuild_git )
 
       endif() ####################################################################################
 
+      if( _PAR_RECURSIVE )
+        ecbuild_info("git submodule --quiet update --init --recursive @ ${ABS_PAR_DIR}")
+        execute_process( COMMAND "${GIT_EXECUTABLE}" submodule --quiet update --init --recursive -q
+                        RESULT_VARIABLE nok ERROR_VARIABLE error
+                        WORKING_DIRECTORY "${ABS_PAR_DIR}")
+        if(nok)
+          ecbuild_warn("git submodule update --init --recursive in ${_PAR_DIR} failed:\n ${error}")
```
Comment (oiffrig) [inline cmake/ecbuild_git.cmake:237]: Is there any particular reason why this should not be a critical failure?

File: cmake/ecbuild_git.cmake [hunk 2]
```diff
@@ -224,6 +228,16 @@ macro( ecbuild_git )
 
       endif() ####################################################################################
 
+      if( _PAR_RECURSIVE )
+        ecbuild_info("git submodule --quiet update --init --recursive @ ${ABS_PAR_DIR}")
+        execute_process( COMMAND "${GIT_EXECUTABLE}" submodule --quiet update --init --recursive -q
```
Comment (oiffrig) [inline cmake/ecbuild_git.cmake:233]:
```suggestion
        execute_process( COMMAND "${GIT_EXECUTABLE}" submodule --quiet update --init --recursive
```

File: cmake/ecbuild_git.cmake (hunk 1 above)
Comment (aerorahul) [reply cmake/ecbuild_git.cmake]: none in particular.  The failure is now critical. Thanks for the review @oiffrig

**Pull request merged**